cnf-certification-test/observability/suite.go

```diff
@@ -151,49 +150,40 @@ func testTerminationMessagePolicy(env *provider.TestEnvironment) {
 	testhelper.AddTestResultLog("Non-compliant", failedContainers, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 
-func checkPDBIsValid(pdb *policyv1.PodDisruptionBudget, replicas *int32) (bool, error) {
-	var replicaCount int32
-	if replicas != nil {
-		replicaCount = *replicas
-	} else {
-		replicaCount = 1 // default value
-	}
-
-	if pdb.Spec.MinAvailable != nil && pdb.Spec.MinAvailable.IntValue() == 0 {
-		return false, fmt.Errorf("field .spec.minAvailable cannot be zero")
-	}
-
-	if pdb.Spec.MaxUnavailable != nil && pdb.Spec.MaxUnavailable.IntValue() >= int(replicaCount) {
-		return false, fmt.Errorf("field .spec.maxUnavailable cannot be greater than or equal to the number of pods in the replica")
-	}
-
-	return true, nil
-}
-
 func testPodDisruptionBudgets(env *provider.TestEnvironment) {
-	failedPDBs := []string{}
-	for i := range env.PodDisruptionBudgets {
-		pdb := &env.PodDisruptionBudgets[i]
-		for pdbLabelKey, pdbLabelValue := range pdb.Spec.Selector.MatchLabels {
-			// Go through all deployments and statefulsets, objects for which PDBs apply
-			for _, deployment := range env.Deployments {
-				if deployment.Spec.Template.Labels[pdbLabelKey] == pdbLabelValue {
-					if ok, err := checkPDBIsValid(pdb, deployment.Spec.Replicas); !ok {
-						failedPDBs = append(failedPDBs, pdb.Name)
-						tnf.ClaimFilePrintf("PDB %s is not valid for deployment %s, err: %v", pdb.Name, deployment.Name, err)
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+
+	// Loop through all of the of Deployments and StatefulSets and check if the PDBs are valid
+	for _, d := range env.Deployments {
+		for k, v := range d.Spec.Template.Labels {
+			for pdbIndex := range env.PodDisruptionBudgets {
+				if env.PodDisruptionBudgets[pdbIndex].Spec.Selector.MatchLabels[k] == v {
+					if ok, err := pdbv1.CheckPDBIsValid(&env.PodDisruptionBudgets[pdbIndex], d.Spec.Replicas); !ok {
+						nonCompliantObjects = append(nonCompliantObjects, testhelper.NewReportObject(fmt.Sprintf("Deployment: %s/%s is missing PodDisruptionBudget", d.Namespace, d.Name), testhelper.DeploymentType, false))
```

Comment:
we are testing deployments and statefulsets, I would create these 2 new types. In the reason for failure of a given deployment/statefulset, maybe add a line per PDB check with PDB name/namespace and saying "OK" or reporting error with "err" output of the CheckPDBIsValid

Reply:
Done.

Reply:
The deployment name, namespace and failure reason are all under failure reason column. Add a Deployment name (add a generic  "name " or "deploymentName" field) and namespace.
![image](https://github.com/test-network-function/cnf-certification-test/assets/86730676/927c9e27-b7b5-4613-acd9-b975d481cfd9)

Reply:
Okay take a look at: https://github.com/test-network-function/cnf-certification-test/pull/1155/commits/258d91ac03fbd58f6d8646daf5a63a5527a889f9 to see if that covers it.